Add SSL setup script for Dockerized Nginx with Let's Encrypt

udkumar opened the pull request:
This commit introduces a new shell script for setting up and managing a Dockerised Nginx environment. Key features include:

- Stopping/starting Nginx container
- Freeing up port 80 on the host
- Installing Nginx and Certbot if not present
- Generating Let's Encrypt SSL certificates
- Setting up automatic SSL certificate renewal
- Copying SSL certificates to a specified Docker volume

The script is intended to streamline the deployment and maintenance of secure Nginx setups in Docker environments.

kavitharaju reviewed (changes requested):
In our existing setup we have nginx and certbot as docker services and we avoid installing them directly on the host machine. Please update this script to work with the docker services. 

How/when to run this shell script? Please document necessary details.

udkumar commented:
I'm currently utilising Docker services exclusively and not using Nginx on the host. 

I've automated the manual processes with a single shell script, which accomplishes the following tasks:

1. Executes our Docker Compose based on the provided steps.
2. Stops the Nginx container.
3. Installs Nginx on the host and sets up SSL for a domain using Let's Encrypt with the aid of Certbot.
4. Transfers the SSL certificate into a volume designated for the Nginx container.
5. Then, it stops the host's Nginx service, which can also be removed if necessary.
6. Finally, it restarts the Nginx container that was initially stopped in step 2

udkumar commented:
@kavitharaju 
I have updated with the following changes:
 

- No manual intervention is required.
- The existing docker compose command (added in readme) will handle the SSL certificate and renew it automatically every 80 days.
- I have modified the shell script to include only the relevant steps.

joelthe1 commented on docker/auto_ssl_script.sh:
```diff
@@ -0,0 +1,73 @@
+fi
+
+# Configuration variables
+NGINX_CONTAINER_NAME="docker-web-server-with-cert-1"
```
Is there a way to not hardcode this and get this from the env?

udkumar replied:
This we can take from env for sure

joelthe1 commented on docker/auto_ssl_script.sh:
```diff
@@ -0,0 +1,73 @@
+generate_certificates() {
+    echo "Generating SSL certificates for $DOMAIN..."
+
+    docker run --rm -it \
```
So does this run docker within docker?

udkumar replied:
Yes, I also checked that the current approach isn't ideal. If it's not suitable, I'm working on creating a single Docker solution, though it's taking some time.

After your comment, I tried but I am getting the same issue as certificate not creating.

joelthe1 commented on docker/docker-compose.yml:
```diff
@@ -267,25 +267,29 @@ services:
+  entrypoint: /bin/sh -c '
+    while :; do
+      /auto_ssl_script.sh;
+      sleep 80d;
```
Is there a way to use the ofelia-scheduler which is already used for scheduled backups for this task?
